pandas/indexes/multi.py

```diff
@@ -371,7 +372,7 @@ def copy(self, names=None, dtype=None, levels=None, labels=None,
             names = self.names
         return MultiIndex(levels=levels, labels=labels, names=names,
                           sortorder=self.sortorder, verify_integrity=False,
-                          _set_identity=_set_identity)
```

Comment:
actually pls remove all fo the multi-index changes and submit in a separate PR this is getting too much / confusing for here.